lawnchair/src/app/lawnchair/search/data/SearchRequest.kt

```diff
@@ -0,0 +1,293 @@
+package app.lawnchair.search.data
+
+import android.content.Context
+import android.database.Cursor
+import android.net.Uri
+import android.provider.ContactsContract
+import android.provider.MediaStore
+import android.provider.Settings
+import android.util.Log
+import app.lawnchair.search.data.suggestion.StartPageService
+import app.lawnchair.util.exists
+import app.lawnchair.util.isDirectory
+import app.lawnchair.util.isHidden
+import app.lawnchair.util.isRegularFile
+import app.lawnchair.util.kotlinxJson
+import app.lawnchair.util.mimeType2Extension
+import com.jakewharton.retrofit2.converter.kotlinx.serialization.asConverterFactory
+import java.lang.reflect.Modifier
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.ResponseBody
+import okio.Path.Companion.toPath
+import org.json.JSONArray
+import org.json.JSONObject
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.create
+
+private val retrofit = Retrofit.Builder()
+    .baseUrl("https://www.startpage.com/")
+    .addConverterFactory(kotlinxJson.asConverterFactory("application/json".toMediaType()))
+    .build()
+
+val startPageService: StartPageService = retrofit.create()
+
+suspend fun getStartPageSuggestions(query: String, max: Int, callback: SearchCallback) {
+    try {
+        if (query.isEmpty() || query.isBlank() || max <= 0) {
+            callback.onSearchLoaded(emptyList())
+            return
+        }
+
+        callback.onLoading()
+
+        withContext(Dispatchers.IO) {
+            val response: Response<ResponseBody> = startPageService.getStartPageSuggestions(
+                query = query,
+                segment = "startpage.lawnchair",
+                partner = "lawnchair",
+                format = "opensearch",
+            )
+
+            if (response.isSuccessful) {
+                val responseBody = response.body()?.string()
+                val suggestions = JSONArray(responseBody).optJSONArray(1)?.let { array ->
```

Comment:
May KotlinX Json API instead.

Reply:
it doesn't need to as it's works as intended